src/OrchardCore/OrchardCore.DisplayManagement/Descriptors/PlacementInfoExtensions.cs

```diff
@@ -22,28 +22,19 @@ public static PlacementInfo Combine(this PlacementInfo first, PlacementInfo seco
             }
             else if (second != null)
             {
-                first.Alternates = first.Alternates.Combine(second.Alternates);
-                first.Wrappers = first.Wrappers.Combine(second.Wrappers);
-                if (!string.IsNullOrEmpty(second.ShapeType))
-                {
-                    first.ShapeType = second.ShapeType;
-                }
+                var combined = new PlacementInfo();
 
-                if (!string.IsNullOrEmpty(second.Location))
-                {
-                    first.Location = second.Location;
-                }
+                combined.Alternates = first.Alternates.Combine(second.Alternates);
+                combined.Wrappers = first.Wrappers.Combine(second.Wrappers);
 
-                if (!string.IsNullOrEmpty(second.DefaultPosition))
-                {
-                    first.DefaultPosition = second.DefaultPosition;
-                }
+                combined.ShapeType = string.IsNullOrEmpty(second.ShapeType) ? first.ShapeType : second.ShapeType;
+                combined.Location = string.IsNullOrEmpty(second.Location) ? first.Location : second.Location;
+                combined.DefaultPosition = string.IsNullOrEmpty(second.DefaultPosition) ? first.DefaultPosition : second.DefaultPosition;
+                combined.Source = $"{first.Source},{second.Source}";
```

Comment:
placements from **placement.json** have null for Source property, so this will results with: ",OrchardCore.Plament" string produced when combined with placements from AdminUI, I would add some additional condition to prevent that

Reply:
Would it be interesting to see where the placement.json is located? As it is now it's possible to see that it's configured in any placement.json file if there is a leading ",". If there is no "," it's only defined in code.

Example: "OrchardCore.Contents,OrchardCore.Placements" or "TheAdmin,OrchardCore.Placements".

In that case I would prefer to do that in a separate Issue/PR.

Reply:
> Would it be interesting to see where the placement.json is located?

in my opinion it might be useful for debugging purposes